bin/setup

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+# This assumes you have general prerequisites installed as by:
+# https://github.com/artsy/potential/blob/master/scripts/setup
+
+# Exit if any subcommand fails
+set -e
+
+echo "brew update and bundle install..."
+brew update
+brew bundle --file=- <<EOF
+brew asdf
+brew node
+brew 'postgresql@12', restart_service: true
+brew 'rabbitmq', restart_service: true
+brew 'yarn'
+EOF
+
+echo "fix rabbitmq stomp plugin not able to bind to port 61613..."
+if ! grep -Fq "stomp.listeners.tcp.1 = 12345" /usr/local/etc/rabbitmq/rabbitmq; then
+  echo "stomp.listeners.tcp.1 = 12345" >> /usr/local/etc/rabbitmq/rabbitmq
+  brew services restart rabbitmq
+fi
+
+echo "Setup asdf, if required..."
+if (! grep 'asdf.sh' ~/.bash_profile)
+then
+  echo -e "\n. $(brew --prefix asdf)/asdf.sh" >> ~/.bash_profile
+  source ~/.bash_profile
+fi
+
+echo "Prepare asdf for erlang and elixir..."
+asdf plugin add erlang
+asdf plugin add elixir
+asdf plugin update erlang
+asdf plugin update elixir
+
+echo "asdf install erlang..."
+# On Mojave (and possibly other Mac versions), you might encounter:
+# configure: error: cannot find required auxiliary files: install-sh config.guess config.sub
+# Installing autoconf 2.69 solves it.
+version=$(sw_vers -productVersion | cut -f1,2 -d'.')
+# if Mojave
+if [[ $version == '10.14.6' ]]
+then
+  brew install autoconf@2.69 && \
+  brew link --overwrite autoconf@2.69 && \
+  autoconf -V
+fi
+asdf install erlang 23.1.1
+asdf local erlang 23.1.1
+
+echo "asdf install elixir..."
+asdf install elixir 1.10.3-otp-23
+asdf local elixir 1.10.3-otp-23
+
+echo "Prepare mix..."
+mix local.hex --force
+mix local.rebar --force
+
+echo "Mix install dependencies..."
+mix deps.get
+
+echo "Install Node.js dependencies..."
+# just use system node/yarn versions, since repo does not specify.
+cd assets && yarn install
+cd ..
+
+echo "Download .env.shared (for common local dev config)..."
+aws s3 cp s3://artsy-citadel/dev/.env.aprd .env.shared
+
+if [ ! -e ".env" ]; then
+  echo "Initialize .env from .env.shared, and from .env.example (for any custom configuration)..."
+  cat .env.shared >> .env
+  echo >> .env
+  cat .env.example >> .env
```

Comment:
Since we don't use `foreman` here, pool all the vars into `.env`.

Reply:
This prevents the script from being re-runnable (to update dependencies or get latest config), right?

This, combined with the manual exporting (`export $(grep -v '^#' .env | xargs)`) instructed below, is sort of unfortunate. Should we just use `foreman` (i.e., the ruby gem) to run an elixir project? That's not ideal, but it would respect both config sources and leave them independently update-able. Maybe a tiny script that just sources from the 2 files in the right order before calling `mix phx.server`? I'm not crazy about that either because it essentially reimplements `foreman` which is likely already on most developers' machines.

Reply:
The script should be re-runnable. If .env already exists, the script skips the entire step, or if .env doesn't exist, it's recreated.

I am not able to find a foreman type of tool for elixir. You mean foreman might actually work with elixir? I hesitate to even try as it's a ruby gem. I go for the wrapper script approach, as that seems better than typing multiple commands to start the server.

Reply:
Yeah `foreman` is a generic wrapper so will work with any subcommands. But your script is nicer.